test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientOpenTypeUpdateTests.cs

```diff
@@ -44,6 +44,20 @@ public void UpdateOpenTypeWithUndeclaredProperties()
             // No more check, this case is to make sure that client doesn't throw exception.
         }
 
+        [TestMethod]
+        public void AddOpenTypeWithUndeclaredProperties()
+        {
+            SetContextWrapper();
+            
+            var row = Row.CreateRow(Guid.NewGuid());
+
+            contextWrapper.Configurations.RequestPipeline.OnEntryStarting(ea => EntryStarting(ea));
+
+            contextWrapper.AddObject("Row", row);
+            contextWrapper.SaveChanges();

```

Comment:
cann't use Assert.NotThrows<...>(....) pattern?

Reply:
Hey @xuzhg I assume you're talking something like [this](https://github.com/OData/odata.net/blob/962aa97d616b34243bbddc9c7bf6cead4a8fe275/test/EndToEndTests/Tests/Client/Build.Desktop/WriteJsonPayloadTests/SerializationInfoEdgeTests.cs#L483). Top of my head, I don't think **Microsoft.VisualStudio.QualityTools.UnitTestFramework** provides `Assert.NotThrows`. Be that as it may, asserting that an exception is not thrown is frowned upon. They actually got rid of it in xUnit - see [this thread](https://github.com/xunit/xunit/issues/188). Any uncaught exception will fail the test hence I'd understand the argument against it. I can however improvise if you feel strongly about it.

Reply:
I'd like to get rid of the current [TestMethod] and use "xunit.net" for all. Spatial, Edm, Core are moved to xunit already.
So, @odero, can you create a task to switch to xunit for the client test cases? 

---
In reply to: [401829651](https://github.com/OData/odata.net/pull/1727#discussion_r401829651) [](ancestors = 401829651)